### PR TITLE
Improve UI and UX

### PR DESCRIPTION
**Pull request by leehuwuj**

### CODE REVIEW

File: admin/components/ui/custom/expandableSection.tsx [hunk 1]
```diff
@@ -1,17 +1,36 @@
...
 }) => {
-  const [isOpen, setIsOpen] = useState(open ?? true);
+  const [isOpen, setIsOpen] = useState<boolean | null>(null);
+
+  // Load the state from local storage when the component mounts
+  useEffect(() => {
+    const savedState = localStorage.getItem("expandable-section-" + name);
+    if (savedState !== null) {
+      setIsOpen(savedState === "open");
+    }
```
Comment (marcusschiesser) [inline admin/components/ui/custom/expandableSection.tsx:22]: Still pass a default ?

File: admin/sections/config/chat.tsx [hunk 2]
```diff
@@ -18,11 +18,16 @@ export const ChatConfig = ({
   form: any;
   isSubmitting: boolean;
 }) => {
+  const chatConfigFields = ["system_prompt", "conversation_starters"];
+  const fieldsChanged = chatConfigFields.some(
+    (field) => form.formState.dirtyFields[field],
```
Comment (marcusschiesser) [inline admin/sections/config/chat.tsx:23]: When is dirty fields updated? On focus lost? Because that's when we want to do the submit

File: admin/sections/config/chat.tsx [hunk 3]
```diff
@@ -57,9 +62,11 @@ export const ChatConfig = ({
           </FormItem>
         )}
       />
```
Comment (marcusschiesser) [inline admin/sections/config/chat.tsx:64]: Idea is to never show the submit button and submit automatically on focus lost

File: admin/sections/config/model.tsx [hunk 4]
```diff
@@ -64,9 +64,13 @@ export const ModelConfig = ({
...
   return (
     <ExpandableSection
-      title={"AI Config"}
-      description="Configure the AI model."
-      open={true}
+      name="update-model"
+      title={values.configured ? "Update model" : "Start"}
+      description={
+        values.configured
+          ? "Change to a different model or other providers"
```
Comment (marcusschiesser) [inline admin/sections/config/model.tsx:71]:
```suggestion
          ? "Change to a different model or use another provider"
```

File: admin/sections/config/model.tsx [hunk 5]
```diff
@@ -64,9 +64,13 @@ export const ModelConfig = ({
...
+          : "Let set up a model to start the app."
```
Comment (marcusschiesser) [inline admin/sections/config/model.tsx:72]:
```suggestion
          : "Set up an AI model to start the app."
```

File: admin/sections/fileLoader.tsx [hunk 6]
```diff
@@ -102,9 +104,11 @@ export const FileLoaderConfig = () => {
               )}
             />
           )}
-          <div className="mt-4">
-            <SubmitButton isSubmitting={false} />
-          </div>
+          {fieldsChanged && (
+            <div className="mt-4">
+              <SubmitButton isSubmitting={false} />
```
Comment (marcusschiesser) [inline admin/sections/fileLoader.tsx:109]: Auto submit on focus lost

File: admin/sections/fileLoader.tsx [hunk 7]
```diff
@@ -26,6 +26,8 @@ export const FileLoaderConfig = () => {
   const loaderForm = useForm({
     resolver: zodResolver(FileLoaderSchema),
   });
+  const fieldsChanged =
+    Object.keys(loaderForm.formState.dirtyFields).length > 0;
```
Comment (marcusschiesser) [inline admin/sections/fileLoader.tsx:30]: Why we can't use this condition in chat.tsx too?

File: admin/sections/config/chat.tsx (hunk 2 above)
Comment (leehuwuj) [reply admin/sections/config/chat.tsx:23]: The form dirty fields are updated once the new value is different from the initial value. I think it makes more sense than on focus lost.

File: admin/sections/config/chat.tsx (hunk 3 above)
Comment (leehuwuj) [reply admin/sections/config/chat.tsx:64]: I still prefer to click Update to submit the changes instead of doing it automatically, because this is admin setting which affects the entire app.

File: admin/sections/fileLoader.tsx (hunk 7 above)
Comment (leehuwuj) [reply admin/sections/fileLoader.tsx:30]: Because we're sharing the same form for model config and chat config but split it into two sections, that's why in chat.tsx we only show the Update button if only the fields in that section has changed

File: admin/components/ui/custom/expandableSection.tsx (hunk 1 above)
Comment (leehuwuj) [reply admin/components/ui/custom/expandableSection.tsx:27]: Great catch! I just update to allow pass as default

File: admin/sections/config/chat.tsx (hunk 3 above)
Comment (marcusschiesser) [reply admin/sections/config/chat.tsx:71]: on the other hand we also already automatically "submit" if we delete or upload a file

File: admin/sections/fileLoader.tsx [hunk 8]
```diff
@@ -64,7 +54,7 @@ export const FileLoaderConfig = () => {
       <Form {...loaderForm}>
         <form
           className="space-y-4 mb-4"
-          onSubmit={loaderForm.handleSubmit(submitLoaderForm)}
+          onBlur={loaderForm.handleSubmit(submitLoaderForm)}
```
Comment (marcusschiesser) [inline admin/sections/fileLoader.tsx:57]: what's the difference between `submitLoaderForm(loaderForm.getValues())` (used below)?

File: admin/sections/config/form.tsx [hunk 9]
```diff
@@ -48,7 +50,10 @@ export const ConfigForm = ({
         setConfigured(false);
       }
       // Reload the chat iframe to apply the changes from the new config
-      demoChatIframeRef.current?.reloadIframe();
+      const changedFields = form.formState.dirtyFields;
+      if (changedFields.conversation_starters) {
+        demoChatIframeRef.current?.reloadIframe();
+      }
```
Comment (marcusschiesser) [inline admin/sections/config/form.tsx:56]: TODO: can be another PR (show reload button if config was changed and do reload only explicitly if the user clicks the reload button)

File: admin/sections/fileLoader.tsx (hunk 8 above)
Comment (leehuwuj) [reply admin/sections/fileLoader.tsx:57]: the `onBlur={loaderForm.handleSubmit(submitLoaderForm)}` is applied to `onBlur` events of the all the fields while `submitLoaderForm(loaderForm.getValues())` is only applied to the current field.

File: admin/sections/config/form.tsx (hunk 9 above)
Comment (leehuwuj) [reply admin/sections/config/form.tsx]: i just added a reload button on top of the iframe, it will always show in the top right. So i think we no longer need auto reload iframe from now.

File: admin/sections/config/chat.tsx (hunk 3 above)
Comment (marcusschiesser) [reply admin/sections/config/chat.tsx:71]: >I still prefer to click Update to submit the changes instead of doing it automatically
is that still the case - what do you suggest @leehuwuj ?

File: admin/sections/knowledge.tsx [hunk 10]
```diff
@@ -16,79 +11,130 @@ import {
...
+      name: file.name,
+      status: "toUpload",
+      blob: file,
+    }));
+    setFiles((prevFiles) => [...prevFiles, ...toUploadFiles]);
+    return toUploadFiles;
+  }
+
+  async function handleUploadFile(toUploadFiles?: File[]) {
+    // Todo
```
Comment (marcusschiesser) [inline admin/sections/knowledge.tsx:72]: what todo?

File: admin/sections/config/chat.tsx [hunk 11]
```diff
@@ -46,7 +45,20 @@ export const ChatConfig = ({
...
+                // Handle for the case when the user deletes a question
+                if (
+                  form.getValues().conversation_starters.length <
+                  field.value?.length
+                ) {
+                  submit(form.getValues(), false);
+                }
+              }}
+            >
               <MultiInput {...field} />
```
Comment (marcusschiesser) [inline admin/sections/config/chat.tsx:62]: can't we just add `onDelete` to `MultiInput` instead?
```suggestion
            >
              <MultiInput {...field} onDelete={() => {
                submit(form.getValues(), false);
              } />
```

File: admin/sections/knowledge.tsx [hunk 12]
```diff
@@ -16,79 +11,129 @@ import {
...
   }
 
+  async function processUpload(blobFiles: globalThis.File[]) {
+    // Ignore duplicate files
+    const duplicateFiles = files.filter((file) =>
+      blobFiles.find((f) => f.name === file.name),
+    );
+    if (duplicateFiles.length > 0) {
+      toast({
+        title: `Duplicate files (${duplicateFiles.map((f) => f.name).join(", ")})!`,
```
Comment (marcusschiesser) [inline admin/sections/knowledge.tsx:121]: we just continue with the upload and ignore duplicate files, right? then how about informing the user? e.g.
```suggestion
        title: `Ignoring duplicate files (${duplicateFiles.map((f) => f.name).join(", ")})!`,
```

File: admin/sections/config/chat.tsx (hunk 11 above)
Comment (leehuwuj) [reply admin/sections/config/chat.tsx]: Great idea, i just added it. Thanks

**Review by marcusschiesser — APPROVED** [review]
